ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,1104 @@
+{
+    "metadata": {
+        "authors": [],
+        "id": "SuppWclYnkyuC51Dlg-6Rg",
+        "kernelspec": {
+            "display_name": "Formula Framework",
+            "language": "C#",
+            "name": "C#"
+        },
+        "language_info": {
+            "file_extension": ".cs",
+            "mimetype": "text/plain",
+            "name": "C#"
+        }
+    },
+    "nbformat": 4,
+    "nbformat_minor": 5,
+    "cells": [
+        {
+            "cell_type": "markdown",
+            "source": [
+                "<a id='report-mutable-scopes'></a>",
+                "\n<p style=\"font-weight:bold;\"> <span style=\"font-size: 36px\"> Report Mutable Scopes</span> </p>"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "This notebook contains the set up of mutable scopes used to achieve high interactivity with reports."
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# References",
+                "\nLibraries and other notebooks which are needed for this notebook are imported below."
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "#!import \"ReportScopes\""
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "using Systemorph.Vertex.Pivot.Builder.Interfaces;",
+                "\nusing Systemorph.InteractiveObjects;",
+                "\nusing Systemorph.Vertex.Session;",
+                "\nusing Systemorph.Vertex.Export;",
+                "\nusing Systemorph.Vertex.Export.Factory;",
+                "\nusing Systemorph.Vertex.InteractiveObjects;"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Form Entity Scopes"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Helper Scopes"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface MutableScopeWithWorkspace<TStorage> : IMutableScopeWithStorage<TStorage> ",
```

Comment:
Why defining this generic scope if at the end the generic type is restricted to one specific type, namely ReportStorage?

Reply:
The same explanation as below